src/Illuminate/Http/Client/PendingRequest.php

```diff
@@ -790,7 +790,7 @@ protected function makePromise(string $method, string $url, array $options = [])
                 });
             })
             ->otherwise(function (TransferException $e) {
-                return $e instanceof RequestException ? $this->populateResponse(new Response($e->getResponse())) : $e;
+                return $e instanceof RequestException && $e->getResponse() ? $this->populateResponse(new Response($e->getResponse())) : $e;
```

Comment:
```suggestion
                return $e instanceof RequestException && $e->hasResponse() ? $this->populateResponse(new Response($e->getResponse())) : $e;
```

I think this should also work

Reply:
Thank you @bogdankharchenko, Good catch!